### PR TITLE
test: add tests for PR #839 review feedback

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary
- Add tests for DeclarationWalker covering catch exception variables, do-while loops, switch/case, and for-each loops (feedback from PR #839 reviews)
- Add CompletionProviderTest for variable shadowing with map literals
- Add @Disabled test documenting multi-class map completion limitation
- Extract shared `stubTypeSolver` helper in CompletionProviderTest
- Fix Kotlin compiler warnings in test files

## Test plan
- [ ] All new tests pass: `./gradlew :semantics-native:test --tests "*DeclarationWalkerTest*"`
- [ ] CompletionProvider tests pass: `./gradlew :groovy-lsp:test --tests "*CompletionProviderTest*"`
- [ ] Full test suite passes

---

> [!NOTE]
> Broadens validation across key subsystems and documents known gaps.
> 
> - Completion: adds map-literal key completion cases (script, method, if, empty map), variable shadowing behavior, and a disabled multi-class case; extracts shared `stubTypeSolver` helper
> - Definition: rewrites `MethodResolutionStrategyTest` with applicability, receiver resolution, method matching (overloads/arg counts), source navigation (with/without sources), and edge cases; several TODO/Disabled tests document limitations
> - Semantic tokens: minor assertion improvements (`assertNotNull`) and position checks for static call/declaration accuracy
> - GDK provider: updates expectations to `parameterTypes`, adds dedup/origin/docs/parameter-name resolution tests and resolver integration, validates indexing across GDK classes
> - Source navigation: small assertion cleanup and method-source fallback behavior test
> - Semantics (native): adds tests for catch variables, do-while, switch cases, and for-each loop in `DeclarationWalker`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 122c1975028e8cf4a6feb2be16260ae8bb1c7e37. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage and organization across multiple test suites with improved setup patterns and helper methods.
  * Added new test scenarios for control structures and method resolution edge cases.
  * Refined assertion patterns for more robust test validation.

*Note: These changes are internal test improvements with no impact on end-user functionality.*

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-lsp/src/test/kotlin/com/github/albertocavalcante/groovylsp/providers/completion/CompletionProviderTest.kt [hunk 1]
```diff
@@ -360,6 +356,71 @@ class CompletionProviderTest {
         )
     }
 
+    @Test
+    fun `should use inner declaration when variable is shadowed`() = runTest {
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-lsp/src/test/kotlin/com/github/albertocavalcante/groovylsp/providers/completion/CompletionProviderTest.kt]: Fixed in c43fd301.
